### PR TITLE
[NUI] Introduce NUI TimePicker

### PR DESCRIPTION
**Pull request by lwc0917**
### Description of Change ###
Introduce TimePicker
https://code.sec.samsung.net/confluence/display/GFX/Picker

### API Changes ###
No API Changes

### CODE REVIEW

**Comment by lwc0917** [comment]
Additional work is required to apply the style of the picker created internally.

**Comment by lwc0917** [comment]
Update time picker style.

File: src/Tizen.NUI.Components/Theme/DefaultTheme.cs [hunk 1]
```diff
@@ -128,6 +128,20 @@ public HashSet<ExternalThemeKeyList> GetExternalThemeKeyListSet()
                     .Add<Size>("/BottomLine/Size", (ViewStyle style, Size value) => ((TabButtonStyle)style).BottomLine.Size = value)
                     .Add<Position>("/BottomLine/Position", (ViewStyle style, Position value) => ((TabButtonStyle)style).BottomLine.Position = value)
                     .AddSelector<Color>("/BottomLine/BackgroundColor", (ViewStyle style, Selector<Color> value) => ((TabButtonStyle)style).BottomLine.BackgroundColor = value, ControlState.Selected),
+                
+                // TimePicker
+                (new ExternalThemeKeyList(typeof(TimePicker), typeof(TimePickerStyle)))
+                    .Add<Size>("/CellPadding", (ViewStyle style, Size value) => ((ViewStyle)style).Size = value)
+                    .Add<Size>("/Pickers/Size", (ViewStyle style, Size value) => ((TimePickerStyle)style).Pickers.Size = value)
```
Comment (jaehyun0cho) [inline src/Tizen.NUI.Components/Theme/DefaultTheme.cs:135]: Please change this as follows.

```
((TimePickerStyle)style).Pickers.Size = value)                 ->                ((PickerStyle)style).Pickers.Size = value)
```

File: src/Tizen.NUI.Components/Theme/DefaultTheme.cs [hunk 2]
```diff
@@ -128,6 +128,20 @@ public HashSet<ExternalThemeKeyList> GetExternalThemeKeyListSet()
                     .Add<Size>("/BottomLine/Size", (ViewStyle style, Size value) => ((TabButtonStyle)style).BottomLine.Size = value)
                     .Add<Position>("/BottomLine/Position", (ViewStyle style, Position value) => ((TabButtonStyle)style).BottomLine.Position = value)
                     .AddSelector<Color>("/BottomLine/BackgroundColor", (ViewStyle style, Selector<Color> value) => ((TabButtonStyle)style).BottomLine.BackgroundColor = value, ControlState.Selected),
+                
+                // TimePicker
+                (new ExternalThemeKeyList(typeof(TimePicker), typeof(TimePickerStyle)))
+                    .Add<Size>("/CellPadding", (ViewStyle style, Size value) => ((ViewStyle)style).Size = value)
```
Comment (jaehyun0cho) [inline src/Tizen.NUI.Components/Theme/DefaultTheme.cs:134]: I am not sure if this is correct.
@rabbitfor Could you let us know if this is correct?

File: src/Tizen.NUI.Components/Theme/DefaultTheme.cs [hunk 3]
```diff
@@ -128,6 +128,20 @@ public HashSet<ExternalThemeKeyList> GetExternalThemeKeyListSet()
...
+                    .Add<Size>("/Pickers/Size", (ViewStyle style, Size value) => ((TimePickerStyle)style).Pickers.Size = value)
+                    .AddSelector<float?>("/Pickers/ItemTextLabel/PixelSize", (ViewStyle style, Selector<float?> value) => ((TimePickerStyle)style).Pickers.ItemTextLabel.PixelSize = value, ControlState.Selected)
```
Comment (jaehyun0cho) [inline src/Tizen.NUI.Components/Theme/DefaultTheme.cs:136]: Please change this as follows.

```
((TimePickerStyle)style).Pickers.ItemTextLabel.PixelSize = value, ControlState.Selected)                ->            ((PickerStyle)style).Pickers.ItemTextLabel.PixelSize = value, ControlState.Selected)
```

File: src/Tizen.NUI.Components/Theme/DefaultTheme.cs [hunk 4]
```diff
@@ -128,6 +128,20 @@ public HashSet<ExternalThemeKeyList> GetExternalThemeKeyListSet()
...
+                    .Add<Size>("/Pickers/ItemTextLabel/Size", (ViewStyle style, Size value) => ((TimePickerStyle)style).Pickers.ItemTextLabel.Size = value)
```
Comment (jaehyun0cho) [inline src/Tizen.NUI.Components/Theme/DefaultTheme.cs:137]: Please change (TimePickerStyle) to (PickerStyle).

File: src/Tizen.NUI.Components/Theme/DefaultTheme.cs [hunk 5]
```diff
@@ -128,6 +128,20 @@ public HashSet<ExternalThemeKeyList> GetExternalThemeKeyListSet()
...
+                    .AddSelector<Color>("/Pickers/ItemTextLabel/TextColor", (ViewStyle style, Selector<Color> value) => ((TimePickerStyle)style).Pickers.ItemTextLabel.TextColor = value, ControlState.Selected)
```
Comment (jaehyun0cho) [inline src/Tizen.NUI.Components/Theme/DefaultTheme.cs:138]: Please change (TimePickerStyle) to (PickerStyle).

**Review by jaehyun0cho — COMMENTED** [review]
Please change (TimePickerStyle) to (PickerStyle) cases in DefaultTheme.cs

File: src/Tizen.NUI.Components/Theme/DefaultTheme.cs (hunk 1 above)
Comment (lwc0917) [reply src/Tizen.NUI.Components/Theme/DefaultTheme.cs]: Pickers is a PickerStyle, so isn't it correct that the style to be referenced is a TimePicker?

File: src/Tizen.NUI.Components/Theme/DefaultTheme.cs (hunk 2 above)
Comment (jaehyun0cho) [reply src/Tizen.NUI.Components/Theme/DefaultTheme.cs:134]: I think this should be changed as follows.
```
((ViewStyle)style).Size = value)          ->            ((TimePicker)style).CellPadding = value)
```

**Review by jaehyun0cho — COMMENTED** [review]
I am so sorry..  you are right.

The original patch was correct.
Please restore the changes.. sorry again.
```
((PickerStyle)style).Pickers          ->         ((TimePickerStyle)style).Pickers
```


Please change the following only.

```
((ViewStyle)style).Size = value)          ->            ((TimePicker)style).CellPadding = value)
```

**Comment by lwc0917** [comment]
@jaehyun0cho  Thanks you for reviewing. i changed PR

**Review by jaehyun0cho — APPROVED** [review]
LGTM

**Pull request merged**